### PR TITLE
Python full version in pipfile

### PR DESCRIPTION
**Pull request by jerempy**
Thank you for contributing to Pipenv!

### The issue
https://github.com/pypa/pipenv/issues/5345

A user requested that when the full version is specified that it gets added to a new pipfile when it is created so that he/she wouldn't need to manually add it themselves.

### The fix

Found the spot where the pipefile [requires] section gets created and modified it slightly to save based on if full version was specified or not.

### The checklist

* [x] Associated issue
* [x] A news fragment in the `news/` directory to describe this fix with the extension `.bugfix.rst`, `.feature.rst`, `.behavior.rst`, `.doc.rst`. `.vendor.rst`. or `.trivial.rst` (this will appear in the release changelog). Use semantic line breaks and name the file after the issue number or the PR #.

### CODE REVIEW

**Review by oz123 — CHANGES_REQUESTED** [review]
Thank you for your PR.

Please fix the linting issue, and add a news fragment.

This is then good to be merged!

**Comment by oz123** [comment]
Please rename the news snippet to:   5345.behavior.rst

**Comment by oz123** [comment]
As I suspected this change leads to breakage of CI.
```
Successfully set up CPython (3.7.14)

```

I think that pinning to a minor version is not a good thing. Unless it's additional. What I mean is, we should have both, e.g.:

```
python_version=3.7
python_full_verison=3.7.14
```

**Comment by jerempy** [comment]
@oz123 I updated it so now the pipfile will record both the python version and the full version in addition if specified. Also updated tests so should work for whatever python version the test is running. Sorry for the back and forth and hopefully we good to go now 🤞

File: news/5345.behavior.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+New pipfiles show python_full_version under [requires] if specified
```
Comment (oz123) [inline news/5345.behavior.rst:1]: Can you please add a longer explanation here?

**Review by oz123 — CHANGES_REQUESTED** [review]

File: news/5345.behavior.rst (hunk 1 above)
Comment (jerempy) [reply news/5345.behavior.rst]: ok cool. good call. Hows it now?

Comment (oz123) [reply news/5345.behavior.rst]: Much better. Thank you.

**Comment by matteius** [comment]
What happens if a new minor version is released?   Will users have to manually update the `python_full_version`?

**Review by matteius — APPROVED** [review]

**Pull request merged**